commitizen/version_schemes.py

```diff
@@ -93,6 +93,24 @@ def micro(self) -> int:
         """The third item of :attr:`release` or ``0`` if unavailable."""
         raise NotImplementedError("must be implemented")
 
+    def __lt__(self, other: Any) -> bool:
```

Comment:
Why were these functions added? I don't see where they are used. Could you add a comment explaining a bit?

- Why are they important?
- When should a developer implement them?

Reply:
The comparators have been added to `VersionProtocol` so that versions can be sorted, e.g. to find the highest version. 

This was actually already used in `bump.Bump.find_previous_final_version`, but the author simply ignored the mypy error:

```python
        final_versions = sorted(final_versions)  # type: ignore
        current_index = final_versions.index(current_version)
        previous_index = current_index - 1
        if previous_index < 0:
            return None
        return final_versions[previous_index]
```

By adding the comparators to the protocol, I was able to remove the `# type: ignore` above.

My change also requires the ability for versions to be sorted, here:

```python
    def get_version(self) -> str:
        matcher = self._tag_format_matcher()
        matches = sorted(
            version
            for t in get_tags(reachable_only=True)
            if (version := matcher(t.name))
        )
        if not matches:
            return "0.0.0"
        return str(matches[-1])
```

I added some tests for sortability.

Reply:
> When should a developer implement them?

since they are part of the `VersionProtocol`, they should always be implemented.  They are required for `bump` and scm provider to work correctly.

Reply:
Thanks for the information. I should've been more clear, I meant a docstring, not a comment on github 😅 My question was aimed more at adding docstrings to help those developers implementing a custom `VersionProtocol`. If you have time to add them in a following PR that would be awesome, but I think we can merge this